### PR TITLE
chore(integration): use consistent naming of `[gG]ithub` in code

### PR DESCRIPTION
**Pull request by pjungermann**
Relates to the discussion at PR #14039.

Relates-to: PR #14039
Relates-to: PR #14174
Signed-off-by: Patrick Jungermann <Patrick.Jungermann@gmail.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by agentbellnorm — APPROVED** [review]
Looks good to me!

File: .changeset/sixty-pigs-shave.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+'@backstage/plugin-catalog-import': patch
+'@backstage/plugin-git-release-manager': patch
+'@backstage/plugin-github-actions': patch
+'@backstage/plugin-github-issues': patch
+'@backstage/plugin-github-pull-requests-board': patch
+'@backstage/plugin-techdocs': patch
+'@backstage/plugin-techdocs-module-addons-contrib': patch
+---
+
+Replaces in-code uses of `GitHub` with `Github` and deprecates old versions.
```
Comment (jhaals) [inline .changeset/sixty-pigs-shave.md:14]: Split this up and use the first part of this changeset for all the `patch` packages and make a separate changeset for the integrations package with the `Deprecates:` section :)

**Review by jhaals — CHANGES_REQUESTED** [review]
awesome

Great job bringing out the 🧹! I wonder if we could add some lint rule that prevents this from happening in the future 🤔 

Small suggestion but otherwise let's 🚢

File: .changeset/sixty-pigs-shave.md (hunk 1 above)
Comment (pjungermann) [reply .changeset/sixty-pigs-shave.md:13]: adjusted

**Review by jhaals — APPROVED** [review]
Thank you!

**Comment by Xantier** [comment]
@pjungermann @jhaals @agentbellnorm Looks like there's a typo in these new names that have been added in. GitHub is written with two capital letters and should be defined camelCased as `gitHub` as defined for example in here: https://google.github.io/styleguide/javaguide.html#s5.3-camel-case

**Comment by pjungermann** [comment]
@Xantier the decision within the backstage codebase was to use `github` and `Github` in code and "GitHub" in next (documentation, changesets, ...). Same applies to similar cases like `gitlab` / `Gitlab` / "GitLab". See also the referenced discussion.
